pkg/ccl/changefeedccl/sink_kafka.go

```diff
@@ -342,12 +343,15 @@ func (s *kafkaSink) EmitRow(
 	if err != nil {
 		return err
 	}
-
+	recordMessageCallback := func(mvcc hlc.Timestamp, bytes int, compressedBytes int) {
+		s.metrics.recordOneMessage()(mvcc, bytes, compressedBytes)
+		s.sinkTelemetry.incEmittedBytes(bytes)
+	}
 	msg := &sarama.ProducerMessage{
 		Topic:    topic,
 		Key:      sarama.ByteEncoder(key),
 		Value:    sarama.ByteEncoder(value),
-		Metadata: messageMetadata{alloc: alloc, mvcc: mvcc, updateMetrics: s.metrics.recordOneMessage()},
+		Metadata: messageMetadata{alloc: alloc, mvcc: mvcc, updateMetrics: recordMessageCallback},
```

Comment:
I can't help but think: can't we change metrics itself to emit to telemetry?  metrics is not really metrics -- it's a metricsRecorder interface... So... just wrap underlying metrics record with "telemetry recorder" implementation.

Or even simpler -- just add direct call to telemetry stuff from inside metrics record.  Either way, I don't see why it's necessary to change every sink implementation.

(And if you absolutely must change every sink implementation, then just write a "telemetrySink" which wraps underlying sink to add telemetry logging).

Reply:
Using the metrics recorder makes sense. Do you have any suggestions for keeping track of time? I liked checking the resolved timestamps in the aggregator because they will be monotonic. Mvcc timestamps from various keys can vary.

Reply:
metric record could keep track of time? Or... add a method?